bgpd/bgp_mplsvpn.c

```diff
@@ -1336,6 +1336,225 @@ leak_update(struct bgp *to_bgp, struct bgp_dest *bn,
 	return new;
 }
 
+void bgp_mplsvpn_path_nh_label_unlink(struct bgp_path_info *path)
+{
+	struct bgp_label_per_nexthop_cache *blnc;
+
+	if (!path)
+		return;
+
+	blnc = path->label_nexthop_cache;
+
+	if (!blnc)
+		return;
+
+	LIST_REMOVE(path, label_nh_thread);
+	path->label_nexthop_cache->path_count--;
+	path->label_nexthop_cache = NULL;
+
+	if (LIST_EMPTY(&(blnc->paths)))
+		bgp_label_per_nexthop_free(blnc);
+}
+
+static int bgp_mplsvpn_get_label_per_nexthop_cb(mpls_label_t label,
+						void *context, bool allocated)
+{
+	struct bgp_label_per_nexthop_cache *blnc = context;
+	mpls_label_t old_label;
+	int debug = BGP_DEBUG(vpn, VPN_LEAK_LABEL);
+	struct bgp_path_info *path;
```

Comment:
`pi` instead of `path`

Reply:
done.

Reply:
also changed ```path``` to ```pi``` for the 'bgp_mplsvpn_path_nh_label_unlink' function.